lib/internal/bootstrap/node.js

```diff
@@ -631,5 +633,92 @@
     new vm.Script(source, { displayErrors: true, filename });
   }
 
+  function setupAllowedFlags() {
+    // This builds process.allowedNodeEnvironmentFlags
+    // from data in the config binding
+
+    const replaceDashesRegex = /-/g;
+    const leadingDashesRegex = /^--?/;
+    const trailingValuesRegex = /=.*$/;
+
+    // Save references so user code does not interfere
+    const replace = Function.call.bind(String.prototype.replace);
+    const has = Function.call.bind(Set.prototype.has);
+    const test = Function.call.bind(RegExp.prototype.test);
+
+    const {
+      allowedV8EnvironmentFlags,
+      allowedNodeEnvironmentFlags
+    } = process.binding('config');
+
+    const trimLeadingDashes = (flag) => replace(flag, leadingDashesRegex, '');
+
+    // Save these for comparison against flags provided to
+    // process.allowedNodeEnvironmentFlags.has() which lack leading dashes.
+    // Avoid interference w/ user code by flattening `Set.prototype` into
+    // each object.
+    const [nodeFlags, v8Flags] = [
+      allowedNodeEnvironmentFlags, allowedV8EnvironmentFlags
+    ].map((flags) => Object.defineProperties(
+      new Set(flags.map(trimLeadingDashes)),
+      Object.getOwnPropertyDescriptors(Set.prototype))
+    );
+
+    class NodeEnvironmentFlagsSet extends Set {
+      constructor(...args) {
+        super(...args);
+
+        // the super constructor consumes `add`, but
+        // disallow any future adds.
+        this.add = () => this;
+      }
+
+      delete() {
+        // noop, `Set` API compatible
```

Comment:
you could also choose to throw here, which would be consistent with `delete` on a frozen object in strict mode.

Reply:
hmm, not a bad idea.

Reply:
I'm going to leave these as-is, I think